### PR TITLE
This Development Cycle in Cargo 1.83

### PR DESCRIPTION
**Pull request by epage**
[Rendered](https://github.com/epage/blog.rust-lang.org/blob/1.83/posts/inside-rust/2024-10-31-this-development-cycle-in-cargo-1.83.md)

### CODE REVIEW

File: posts/inside-rust/2024-10-31-this-development-cycle-in-cargo-1.83.md
Comment (epage) [inline posts/inside-rust/2024-10-31-this-development-cycle-in-cargo-1.83.md:1]: > posts/inside-rust/2024-10-31-this-development-cycle-in-cargo-1.83.md

I put this for publishing tomorrow

File: posts/inside-rust/2024-10-31-this-development-cycle-in-cargo-1.83.md [hunk 1]
```diff
@@ -0,0 +1,351 @@
...
+
+*Update from [1.81](https://blog.rust-lang.org/inside-rust/2024/08/15/this-development-cycle-in-cargo-1.81.html#optimizing-git-sources)*
+
+On [zulip](https://rust-lang.zulipchat.com/#narrow/channel/246057-t-cargo/topic/.60cargo.20metadata.60.20performance),
+some tool authors were discussing with the Cargo team how to optimize `cargo metadata`.
+
+One of the concerns raised was with the serialization / deserialization time of json.
+In considering supporting another format, there has to be enough of a difference from json to be worth the effort, it must have a good compatibility story, and we should consider the interoperability of the format.
+For example, CBOR was brought up.
+While it has an IETFC RFC,
```
Comment (weihanglo) [inline posts/inside-rust/2024-10-31-this-development-cycle-in-cargo-1.83.md:149]: Is it a typo?
```suggestion
While it has an IETF RFC,
```

File: posts/inside-rust/2024-10-31-this-development-cycle-in-cargo-1.83.md [hunk 2]
```diff
@@ -0,0 +1,351 @@
...
+With that out of the way,
+we used the process in
+[#12432](https://github.com/rust-lang/cargo/issues/12432)
+to officially accept a build script API package.
+
+## Misc
+
+- [Daily reports](https://rust-lang.zulipchat.com/#narrow/stream/260232-t-cargo.2FPubGrub/topic/Progress.20report) by [Eh2406](https://github.com/Eh2406) on the progress of the Rust implementation of the PugGrub version solving algorithm
+- [Flowrey](https://github.com/Flowrey) finished their work on a `--dry-run` flag for `cargo install` ([#14280](https://github.com/rust-lang/cargo/pull/14280))
+- [ahaoboy](https://github.com/ahaoboy) added dark mode to `--iming` ([#](https://github.com/rust-lang/cargo/pull/14588))
```
Comment (weihanglo) [inline posts/inside-rust/2024-10-31-this-development-cycle-in-cargo-1.83.md:283]:
```suggestion
- [ahaoboy](https://github.com/ahaoboy) added dark mode to `--timings` ([#14588](https://github.com/rust-lang/cargo/pull/14588))
```

**Review by weihanglo — APPROVED** [review]

File: posts/inside-rust/2024-10-31-this-development-cycle-in-cargo-1.83.md [hunk 3]
```diff
@@ -0,0 +1,351 @@
...
+- [Misc](#misc)
+- [Focus areas without progress](#focus-areas-without-progress)
+
+## Plugin of the cycle
+
+Cargo can't be everything to everyone,
+if for no other reason than the compatibility guarantees it must uphold.
+Plugins play an important part of the Cargo ecosystem and we want to celebrate them.
+
+Our plugin for this cycle is [`cargo-bloat`](https://crates.io/crates/cargo-bloat) what is taking up space in your binary.
```
Comment (Kriskras99) [inline posts/inside-rust/2024-10-31-this-development-cycle-in-cargo-1.83.md:33]:
```suggestion
Our plugin for this cycle is [`cargo-bloat`](https://crates.io/crates/cargo-bloat) which shows what is taking up space in your binary.
```

File: posts/inside-rust/2024-10-31-this-development-cycle-in-cargo-1.83.md [hunk 4]
```diff
@@ -0,0 +1,351 @@
...
+[#14620](https://github.com/rust-lang/cargo/pull/14620),
+[#14662](https://github.com/rust-lang/cargo/pull/14662),
+)
+and wrote down the lessons learned on choosing an MSRV from the RFC process
+([#14636](https://github.com/rust-lang/cargo/pull/14636)).
+
+Previous, we discussed suboptimal results when resolving dependencies with multiple distinct MSRVs in a workspace.
+We were resolving for the lowest MSRV in the workspace.
+For a workspace member with any other MSRV,
+they would either get a version older than needed for too new.
```
Comment (Kriskras99) [inline posts/inside-rust/2024-10-31-this-development-cycle-in-cargo-1.83.md:57]:
```suggestion
they would either get a version older than needed or too new.
```

**Review by Kriskras99 — COMMENTED** [review]
Thanks for writing these, they're very informative!

I found two sentences that didn't seem entirely correct.

**Review by ehuss — APPROVED** [review]
Thanks!

**Pull request merged**